src/main/java/org/terasology/staticCities/walls/TownWallFacetProvider.java

```diff
@@ -101,7 +104,8 @@ public void process(GeneratingRegion region) {
         for (Settlement settlement : settlementFacet.getSettlements()) {
             if (settlement.hasTownwall()) {
                 Site site = settlement.getSite();
-                if (Circle.intersects(site.getPos(), site.getRadius(), wallFacet.getWorldRegion())) {
+                Rectanglef bounds = wallFacet.getWorldRegion().getBounds(new Rectanglef());
```

Comment:
We do have some intersection checks directly on BlockRegion - should we also add something like `BlockArea::intersectsCircle(...)` in a similar matter?